packages/web/src/components/change-email/ChangeEmailModal.tsx

```diff
@@ -0,0 +1,249 @@
+    </TextLink>
+  )
+}
+
+const CurrentEmail = () => {
+  const [{ value: oldEmail }, , { setValue: setOldEmail }] =
+    useField('oldEmail')
+  // Load the email for the user
+  const emailRequest = useAsync(audiusBackendInstance.getUserEmail)
+  useEffect(() => {
+    if (emailRequest.value) {
+      setOldEmail(emailRequest.value)
+    }
+  }, [emailRequest.loading, emailRequest.value, setOldEmail])
+  return (
+    <Text variant='body' size='m'>
+      {oldEmail !== '' ? (
+        oldEmail
+      ) : (
+        <LoadingSpinner className={styles.inlineSpinner} />
+      )}
+    </Text>
+  )
+}
+
+export const ConfirmPasswordPage = () => {
+  return (
+    <Flex direction='column' gap='xl'>
+      <Text variant='body'>{messages.confirmPasswordHelp}</Text>
+      <Box>
+        <Text variant='label' size={'xs'}>
+          {messages.currentEmail}
+        </Text>
+        <CurrentEmail />
+      </Box>
+      <HarmonyPasswordField
+        name='password'
+        label={messages.currentPassword}
+        autoComplete={'password'}
+      />
+    </Flex>
+  )
+}
+
+export const NewEmailPage = () => {
+  return (
+    <Flex direction='column' gap='xl'>
+      <Text variant='body'>{messages.newEmailHelp}</Text>
+      <Box>
+        <Text variant='label' size={'xs'}>
+          {messages.currentEmail}
+        </Text>
+        <CurrentEmail />
+      </Box>
+      <HarmonyTextField name='email' label={messages.newEmail} />
+    </Flex>
+  )
+}
+
+export const VerifyEmailPage = () => {
+  return (
+    <Flex direction='column' gap='xl'>
+      <Text variant='body'>{messages.verifyEmailHelp}</Text>
+      <HarmonyTextField
+        name={'otp'}
+        label={messages.code}
+        placeholder={messages.otpPlaceholder}
+        transformValueOnChange={formatOtp}
+      />
+      <Text variant='body'>
+        {messages.resendHelp}
+        <ResendCodeLink />
+      </Text>
+    </Flex>
+  )
+}
+
+export const SuccessPage = () => {
+  const [{ value: oldEmail }] = useField('oldEmail')
+  const [{ value: email }] = useField('email')
+  return (
+    <Text variant={'body'}>
+      {messages.success(oldEmail)}
+      <Text asChild strength={'strong'}>
+        <span>{email}</span>
+      </Text>
+      !
+    </Text>
+  )
+}
+
+const ChangeEmailModalForm = ({
+  onClose,
+  page
+}: Pick<ChangeEmailModalProps, 'onClose'> & {
+  page: ChangeEmailPage
+}) => {
+  const { isSubmitting } = useFormikContext()
+
+  const isSuccessPage = page === ChangeEmailPage.Success
+
+  return (
+    <ModalForm>
+      <ModalHeader onClose={onClose}>
+        <ModalTitle title={messages.title} icon={<IconKey />} />
+      </ModalHeader>
+      <ModalContentPages currentPage={page}>
+        <ConfirmPasswordPage />
+        <NewEmailPage />
+        <VerifyEmailPage />
+        <SuccessPage />
+      </ModalContentPages>
+      <ModalFooter className={styles.footer}>
+        {isSuccessPage ? (
+          <Button type='button' fullWidth onClick={onClose}>
```

Comment:
type button is needed here?

Reply:
I think so or it submits? figured I'd be explicit